Submitted By: Archaic (archaic -aT- linuxfromscratch -DoT- org)
Date: 2005-04-27
Initial Package Version: 2.3.4
Origin: http://sources.redhat.com/ml/libc-alpha/2005-04/msg00056.html
Upstream Status: Applied
Description: Allows tst-cancel17 and tst-cancelx17 to pass with kernel-2.6.11.x

diff -Naur glibc-2.3.5.orig/nptl/tst-cancel17.c glibc-2.3.5/nptl/tst-cancel17.c
--- glibc-2.3.5.orig/nptl/tst-cancel17.c	2003-08-08 02:09:57.000000000 -0500
+++ glibc-2.3.5/nptl/tst-cancel17.c	2005-04-27 10:46:26.000000000 -0500
@@ -1,4 +1,4 @@
-/* Copyright (C) 2003 Free Software Foundation, Inc.
+/* Copyright (C) 2003, 2005 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
    Contributed by Ulrich Drepper <drepper@redhat.com>, 2003.
 
@@ -228,7 +228,7 @@
 
   size_t len2 = fpathconf (fds[1], _PC_PIPE_BUF);
   size_t page_size = sysconf (_SC_PAGESIZE);
-  len2 = (len2 < page_size ? page_size : len2) + sizeof (mem) + 1;
+  len2 = 20 * (len2 < page_size ? page_size : len2) + sizeof (mem) + 1;
   char *mem2 = malloc (len2);
   if (mem2 == NULL)
     {
